src/sentry/uptime/detectors/tasks.py

```diff
@@ -0,0 +1,169 @@
+import datetime
+from datetime import timedelta
+
+from django.utils import timezone
+
+from sentry.locks import locks
+from sentry.models.project import Project
+from sentry.tasks.base import instrumented_task
+from sentry.uptime.detectors.ranking import (
+    _get_cluster,
+    delete_candidate_urls_for_project,
+    delete_project_bucket,
+    get_candidate_urls_for_project,
+    get_project_bucket,
+)
+from sentry.uptime.models import ProjectUptimeSubscription, UptimeSubscription
+from sentry.utils import metrics
+from sentry.utils.locking import UnableToAcquireLock
+from sentry_plugins.jira.client import md5
+
+LAST_PROCESSED_KEY = "uptime_detector_last_processed"
+SCHEDULER_LOCK_KEY = "uptime_detector_scheduler_lock"
+FAILED_URL_RETRY_FREQ = timedelta(days=7)
+URL_MIN_TIMES_SEEN = 5
+URL_MIN_PERCENT = 0.05
+
+
+@instrumented_task(
+    name="sentry.uptime.detectors.tasks.schedule_detections",
+    queue="uptime",
+    time_limit=60,
+    soft_time_limit=55,
+)
+def schedule_detections():
+    """
+    Runs regularly and fires off a task for each detection bucket that needs to be run since
+    this task last ran.
+    """
+    lock = locks.get(
+        SCHEDULER_LOCK_KEY,
+        duration=60,
+        name="uptime.detection.schedule_detections",
+    )
+    try:
+        with lock.acquire():
+            cluster = _get_cluster()
+            last_processed = cluster.get(LAST_PROCESSED_KEY)
+            if last_processed is None:
+                last_processed = timezone.now().replace(second=0, microsecond=0)
+            else:
+                last_processed = datetime.datetime.fromtimestamp(
+                    int(last_processed), tz=datetime.UTC
+                )
+
+            minutes_since_last_processed = int(
+                (timezone.now() - last_processed) / timedelta(minutes=1)
+            )
```

Comment:
This is probably more clear, but you can use the `//` integer division operator too

```suggestion
            minutes_since_last_processed = (timezone.now() - last_processed) // timedelta(minutes=1)
```

yeah way less clear lol